.github/workflows/api-unit-tests.yml

```diff
@@ -0,0 +1,33 @@
+name: API unit tests
+
+on:
```

Comment:
Don't you need also the `workflow_dispatch` case? This will allow to run them manually, which can be useful to run nightly executions.